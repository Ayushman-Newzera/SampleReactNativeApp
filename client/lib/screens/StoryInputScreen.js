var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _react=_interopRequireDefault(require("react"));var _reactNative=require("react-native");var _TextInput=_interopRequireDefault(require("../utilities/TextInput"));var _SimpleImagePicker=_interopRequireDefault(require("./SimpleImagePicker"));var _jsxFileName="/Users/ayushman/Desktop/SampleReactNativeApp1/client/src/screens/StoryInputScreen.js";function StoryInputScreen(_ref){var navigation=_ref.navigation;var handleNavigation=function handleNavigation(){navigation.pop();};return _react.default.createElement(_reactNative.View,{style:styles.container,__self:this,__source:{fileName:_jsxFileName,lineNumber:13,columnNumber:5}},_react.default.createElement(_SimpleImagePicker.default,{asyncKey:"@storyPicture",handleNavigation:handleNavigation,__self:this,__source:{fileName:_jsxFileName,lineNumber:14,columnNumber:7}}),_react.default.createElement(_TextInput.default,{__self:this,__source:{fileName:_jsxFileName,lineNumber:19,columnNumber:7}}),_react.default.createElement(_reactNative.View,{style:styles.submitButton,__self:this,__source:{fileName:_jsxFileName,lineNumber:21,columnNumber:7}},_react.default.createElement(_reactNative.Button,{title:"Submit",onPress:console.log('Button Clicked'),__self:this,__source:{fileName:_jsxFileName,lineNumber:22,columnNumber:9}})));}var styles=_reactNative.StyleSheet.create({container:{position:'relative',flex:1},buttonContainer:{position:'relative',flexDirection:'row',alignItems:'center',justifyContent:'center',marginTop:50},lowerButton:{position:'relative',marginLeft:50},submitButton:{position:'relative',width:150,marginTop:50,marginLeft:120,marginBottom:50}});var _default=StoryInputScreen;exports.default=_default;