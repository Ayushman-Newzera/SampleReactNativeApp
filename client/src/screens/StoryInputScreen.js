// @flow

import React from 'react';
import {StyleSheet, View} from 'react-native';
import SimpleImagePicker from './SimpleImagePicker';

/**
 *
 * @param {navigation} it is the prop to handle navigation
 * @returns StoryInput Screen
 */
function StoryInputScreen({navigation}) {
  /** Function to handle navigation */
  const handleNavigation = () => {
    navigation.navigate('Home');
  };

  return (
    <View style={styles.container}>
      <SimpleImagePicker
        asyncKey="@storyPicture"
        handleNavigation={handleNavigation}
        title="Pick the image for your story"
      />

      {/* currently we are not using the below functionality  */}

      {/* <UselessTextInputMultiline /> */}

      {/* <View style={styles.submitButton}>
        <Button title="Submit" onPress={() => navigation.navigate('Home')} />
      </View> */}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    position: 'relative',
    flex: 1,
  },
  buttonContainer: {
    position: 'relative',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 50,
  },
  lowerButton: {
    position: 'relative',
    marginLeft: 50,
  },
  submitButton: {
    position: 'relative',
    width: 150,
    marginTop: 50,
    marginLeft: 120,
    marginBottom: 50,
  },
});

export default StoryInputScreen;
