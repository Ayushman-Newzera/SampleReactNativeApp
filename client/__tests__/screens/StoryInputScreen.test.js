import {shallow} from 'enzyme';
import React from 'react';
import StoryInputScreen from '../../src/screens/StoryInputScreen';

/**
 * Used to create an object of props
 * @param {object} props Custom props that needs to pass
 * @returns Props to be passed to the component/screen
 */
const createTestProps = props => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

/** Following are the tests for StoryInputScreen */
describe('StoryInputScreen', () => {
  /** Snapshot Testing */
  it('Component should render correctly', () => {
    const component = shallow(<StoryInputScreen />);
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });

  /** Navigation should be handled properly */
  it('HandleNavigation function should work correctly', async () => {
    let props = createTestProps({});
    let wrapper = shallow(<StoryInputScreen {...props} />);
    const SimpleImagePicker = wrapper.find('SimpleImagePicker').get(0);
    await SimpleImagePicker.props.handleNavigation();
    expect(props.navigation.navigate).toHaveBeenCalledWith('Home');
  });
});
