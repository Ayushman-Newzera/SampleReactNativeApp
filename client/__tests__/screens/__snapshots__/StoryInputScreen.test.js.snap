// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryInputScreen Component should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <SimpleImagePicker
    asyncKey="@storyPicture"
    handleNavigation={[Function]}
    title="Pick the image for your story"
  />
</View>
`;
